bpf/kmesh/workload/sockops.c

```diff
@@ -56,14 +60,28 @@ static inline void extract_skops_to_tuple(struct bpf_sock_ops *skops, struct bpf
         tuple_key->ipv4.sport = bpf_htons(GET_SKOPS_LOCAL_PORT(skops));
         // remote_port is network byteorder
         tuple_key->ipv4.dport = GET_SKOPS_REMOTE_PORT(skops);
-    } else {
+    }
+    if (skops->family == AF_INET6) {
         bpf_memcpy(tuple_key->ipv6.saddr, skops->local_ip6, IPV6_ADDR_LEN);
         bpf_memcpy(tuple_key->ipv6.daddr, skops->remote_ip6, IPV6_ADDR_LEN);
         // local_port is host byteorder, need to htons
         tuple_key->ipv6.sport = bpf_htons(GET_SKOPS_LOCAL_PORT(skops));
         // remote_port is network byteorder
         tuple_key->ipv6.dport = GET_SKOPS_REMOTE_PORT(skops);
     }
+
+    if (is_ipv4_mapped_addr(tuple_key->ipv6.daddr)) {
```

Comment:
I doubt this is not right. If the dst address is mapped address.  Assume you want to extract to ipv4, i am not sure why operating on ip6 below, since is it a union of ipv4 and ipv6.

And BTW, since the family == AF_INET6, how can we make it work when we dnat to v4?

Can you test with a real example